cves/kernel/CVE-2015-7566.yml

```diff
@@ -55,7 +55,9 @@ description_instructions: |
 
   Your target audience is people just like you before you took any course in
   security
-description:
+description: |
+  When using a USB device that lacks a bulk-out endpoint (what sends data from the host to the device), 
+  a NULL pointer error occurs. This causes the system to crash which can lead to more errors and corruption.
```

Comment:
I think more background would be nice in this. Where was it introduced? What kind of data is the host sending to the device that is causing this crash?

Reply:
Found the how of the vulnerability under:
https://bugzilla.redhat.com/show_bug.cgi?id=1283371

-----
```
Description:
The bug was found using the USB-fuzzing framework vUSBf from Sergej Schumilo 
(github.com/schumilo) using the following device descriptor:

 ######### PAYLOAD 1 #########
[*] Device-Descriptor
  bLength:		0x12
  bDescriptorType:	0x1
  bcdUSB:		0x200
  bDeviceClass:		0x3
  bDeviceSubClass:	0x0
  bDeviceProtocol:	0x0
  bMaxPacketSize:	0x40
  idVendor:		0x54c
  idProduct:		0x144
  bcdDevice:		0x100
  iManufacturer:	0x1
  iProduct:		0x2
  iSerialNumbers:	0x3
  bNumConfigurations:	0x1

The clie_5_attach function of the visor driver, which is called during the driver initialization process, expects an OUT-Bulk-Endpoint. 
Due to an incomplete sanity check, the visor driver tries to dereference null-pointers. 
This results in a crash of the system.
```